### PR TITLE
Update to Freedesktop SDK 22.08

### PR DESCRIPTION
**Pull request by quyykk**
This PR simply updates the SDK version to 22.08, mainly for the latest version of SDL.

The scons build rules broke when I updated the SDK version, so I used `flatpak-pip-generator` to create a separate json.

### CODE REVIEW

File: io.github.endless_sky.endless_sky.json [hunk 1]
```diff
@@ -1,40 +1,23 @@
...
-            ],
-            "sources": [
-                {
-                    "type": "archive",
-                    "url": "https://github.com/scons/scons/archive/4.1.0.tar.gz",
-                    "sha512": "f79b86bb09783767b3872cfb8efb665372714a604af2aaf3adc66eee63d3afe27bc6b2aab83813743c83f71c81c800d42842e916501787ba402ce2726dda9b44"
-                }
-            ]
-        },
+		"python3-scons.json",
```
Comment (Pointedstick) [inline io.github.endless_sky.endless_sky.json:20]: Indentation is off (we use spaces, not tabs)

**Comment by quyykk** [comment]
I'm not sure what's wrong with this, but the game doesn't start anymore:

```
(SDL message: "No available video device")
```

File: io.github.endless_sky.endless_sky.json [hunk 2]
```diff
@@ -1,40 +1,23 @@
...
     "sdk": "org.freedesktop.Sdk",
     "command": "endless-sky",
     "rename-desktop-file": "endless-sky.desktop",
     "rename-icon": "endless-sky",
     "finish-args": [
         "--socket=pulseaudio",
         "--socket=wayland",
-        "--socket=x11",
+        "--socket=fallback-x11",
+        "--share=ipc",
```
Comment (Pointedstick) [inline io.github.endless_sky.endless_sky.json:13]: These a good changes, but not relevant to this PR. Can you do them in another one?

File: python3-scons.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Pointedstick) [inline python3-scons.json:1]: Moving this to a separate file doesn't seem necessary; why can't we keep it inline?

File: python3-scons.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "name": "python3-scons",
+    "buildsystem": "simple",
+    "build-commands": [
+        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"scons\" --no-build-isolation"
+    ],
+    "sources": [
+        {
+            "type": "file",
+            "url": "https://files.pythonhosted.org/packages/1a/e1/7fad50e1baa7068e3e2382b11c4d00190238798d62bfa6bbfa8619c19f39/SCons-4.4.0-py3-none-any.whl",
```
Comment (Pointedstick) [inline python3-scons.json:10]: We shouldn't change the location we're getting `scons` from. It should remain an official archive location.

**Review by Pointedstick — CHANGES_REQUESTED** [review]

File: io.github.endless_sky.endless_sky.json (hunk 2 above)
Comment (quyykk) [reply io.github.endless_sky.endless_sky.json:13]: sure thing

File: python3-scons.json (hunk 3 above)
Comment (quyykk) [reply python3-scons.json:1]: idk I followed the guidelines. I'll keep it inline.

File: python3-scons.json (hunk 4 above)
Comment (quyykk) [reply python3-scons.json:10]: That's the official package location for `pip` packages. Is that still fine?

Comment (Pointedstick) [reply python3-scons.json:10]: We shouldn't use `pip` during a build process at all. :) We should download the source from its official location, as we currently do.

**Comment by quyykk** [comment]
> I'm not sure what's wrong with this, but the game doesn't start anymore:
> 
> ```
> (SDL message: "No available video device")
> ```

Seems to be a glew issue :confused: Yet another reason to stop using glew.

**Comment by Pointedstick** [comment]
Maybe fixed with the change in the new release? Maybe we should wait until Anitya notices it and updates the manifest, and then it'll be fixed automatically.

**Comment by quyykk** [comment]
Unfortunately no. I just tried building 0.9.15 manually and yup, same error.

File: python3-scons.json (hunk 4 above)
Comment (quyykk) [reply python3-scons.json:10]: But it's way easier to install python packages through pip instead of installing them manually :)

Comment (Pointedstick) [reply python3-scons.json:10]: Like I said, it's not good practice to do something like that during a formal build process. It's volatile, things can change from under you... it's just not accepted practice for high quality build system code.

**Comment by quyykk** [comment]
Good news, it seems like it's not a glew issue after all! I can't run the flatpak if I build it myself, but if I install the test build above it works! (but since this is 0.9.14 on Wayland, it fails with that glew error).

**Comment by Pointedstick** [comment]
bot, build.

**Review by Pointedstick — APPROVED** [review]
Thanks!

**Pull request merged**